### ISSUE TITLE
Duplicate calendar events

### ISSUE THREAD

**Issue by LnLcFlx** [IgnisDa/ryot#446]
After #426 was fixed, many duplicates of movie release dates started appearing. First some movies only had the same release date twice, then triple ...

I don't know if these are actually dates on TMDB or just the same redundant dates.

![Screenshot_20231027-094535_Fennec](https://github.com/IgnisDa/ryot/assets/22853309/a8254f7c-2c31-4533-9ad0-a7a3174cd888)

**Comment by IgnisDa** [comment]
Thanks for reporting this. Happened on my instance too, but I did not notice it. I will fix it.

**Comment by IgnisDa** [comment]
Blocked by https://github.com/SeaQL/sea-query/pull/478.

Note to self create 2 partial indices:
- when metadata_extra_information is `NULL`
- when it is `NOT NULL`

**Comment by IgnisDa** [comment]
@LnLcFlx This will likely take some time to be fixed. For the time being, if it is really bothering you, you can run this SQL to remove duplicate elements.

```sql
CREATE OR REPLACE FUNCTION process_event_ids() RETURNS VOID AS $$
DECLARE
    rec record; id2 integer;
BEGIN
    FOR rec IN
        SELECT date, metadata_id, ARRAY_AGG(id) AS event_ids
        FROM calendar_event
        WHERE metadata_extra_information IS NULL
        GROUP BY date, metadata_id
        HAVING COUNT(*) > 1
    LOOP
        FOREACH id2 IN ARRAY rec.event_ids[2:]
        LOOP
            RAISE NOTICE 'Processing event_id %', id2;
            DELETE FROM calendar_event where id = id2;
        END LOOP;
    END LOOP;
END;
$$ LANGUAGE plpgsql;
select process_event_ids();
```

It worked on my instance, but make sure you backup your DB just in case.

**Comment by IgnisDa** [comment]
@LnLcFlx can you confirm if everything works as expected now?

**Comment by LnLcFlx** [comment]
Prior to updating, I have deleted all duplicates using your SQL.
But so far I did not encounter any new duplicates.
Will report if it still happens.
